### PR TITLE
Jim/juris filter

### PR DESCRIPTION
**Pull request by jimduff-usds**
This PR adds JurisdictionalFilters, which enable us to write complex filters in code.  This was necessitated by the current county filtering needs.

## Checklist
- [X] Tested locally?
- [ ] Added new dependencies?
- [ ] Updated the docs?
- [X] Added a test?
- [X] Did you check for sensitive data, and remove any?
- [ ] Are additional approvals needed for this change?
- [ ] Are there potential vulnerabilities or licensing issues with any new dependencies introduced?

## Fixes 

- #138 
- #136 

## To Be Done

- Decide if its ok that Tablesaw "leaks" into JurisdictionFilters.  The Selection feature of Tablesaw was so powerful and general, it seemed silly not to use it.

### CODE REVIEW

File: prime-router/metadata/organizations-local.yml [hunk 1]
```diff
@@ -16,7 +16,8 @@
       - name: elr
         topic: covid-19
         schema: az/az-covid-19
-        jurisdictionalFilter: { standard.testing_lab_state: AZ }
+        jurisdictionalFilter:
+          - matches(standard.ordering_facility_state,AZ)
```
Comment (jimduff-usds) [inline prime-router/metadata/organizations-local.yml:20]: Throughout the organization.yml files, the new `matches(column, value) `replaces the old   `column: value` syntax.

File: prime-router/src/main/kotlin/Element.kt [hunk 2]
```diff
@@ -247,22 +247,31 @@ data class Element(
...
                     OffsetDateTime.parse(formattedValue, formatter)
                 } catch (e: DateTimeParseException) {
                     null
-                    // Finally, accept a local date pattern and assume it is in the central timezone (small amount of error)
                 } ?: try {
+                    // Try to parse using a LocalDate pattern assuming it is in our canonical dateFormatter. Central timezone.
                     val date = LocalDate.parse(formattedValue, dateFormatter)
                     val zoneOffset = ZoneId.of(USTimeZone.CENTRAL.zoneId).rules.getOffset(Instant.now())
                     OffsetDateTime.of(date, LocalTime.of(0, 0), zoneOffset)
+                } catch (e: DateTimeParseException) {
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/Element.kt:266]: Obscure bug fix when reading yyyy-MM-dd data back into the objects.

File: prime-router/src/main/kotlin/JurisdictionalFilters.kt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            ?: error("Unable to filterByCounty:  column standard.patient_state not found.")
+        val patientCounty = table.stringColumn("standard.patient_county")
+            ?: error("Unable to filterByCounty:  column standard.patient_county not found.")
+        val facilityState = table.stringColumn("standard.ordering_facility_state")
+            ?: error("Unable to filterByCounty:  column standard.ordering_facility_state not found.")
+        val facilityCounty = table.stringColumn("standard.ordering_facility_county")
+            ?: error("Unable to filterByCounty:  column standard.ordering_facility_county not found.")
+
+        // Try to be very loose on county matching.   Anything with the county name embedded is ok.
+        val countyRegex = "(?i).*${args[1]}.*"
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/JurisdictionalFilters.kt:74]: If we implement strong validation on county names, then we can make this a much tighter matching.

File: prime-router/src/main/kotlin/JurisdictionalFilters.kt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+object JurisdictionalFilters {
+    /**
+     * filterFunction must be of form "funName(arg1, arg2, etc)"
+     */
+    fun parseJurisdictionalFilter(filterFunction: String): Pair<String, List<String>> {
+// REMOVE        val match = Regex("([a-zA-Z0-9]+)\\x28([a-z, \\x2E_\\x2DA-Z0-9]*)\\x29").find(filterFunction)
+        // Using a permissive match in the (arg1, arg2) section, to allow most regexs to be passed as args.
+        // Somehow this works with "(?i).*Pima.*", I guess because the \\x29 matches rightmost ')' char
+        val match = Regex("([a-zA-Z0-9]+)\\x28(.*)\\x29").find(filterFunction)
```
Comment (jimduff-usds) [inline prime-router/src/main/kotlin/JurisdictionalFilters.kt:91]: The generic (.*) in the middle was necessitated by the need to pass gloppy regex'es in.

**Review by MauriceReeves-usds — APPROVED** [review]

File: prime-router/metadata/schemas/AZ/pima-az-covid-19-csv.schema [hunk 5]
```diff
@@ -26,8 +26,8 @@ elements:
     type: TEXT
 
   # LOINC code for the test performed
-  - name: standard.test_performed_code
-    csvFields: [{name: Ordered_test_code, format: $display}]
+  - name: standard.test_performed_name
+    csvFields: [{name: Ordered_test_name}]
```
Comment (jimduff-usds) [inline prime-router/metadata/schemas/AZ/pima-az-covid-19-csv.schema:30]: Late change to fix Pima issue

**Review by RickHawesUSDS — APPROVED** [review]
Great work.

**Pull request merged**